dashboard/app/controllers/openai_controller.rb

```diff
@@ -0,0 +1,34 @@
+class OpenaiController < ApplicationController
+  # POST /openai/chat_completion
+  def chat_completion
+    return unless current_user.ai_chat_access?
```

Comment:
Permissions for controller actions are typically managed in `ability.rb` using a gem called CanCanCan -- I think you'd want to add something like this, rather than using an early return:

https://github.com/code-dot-org/code-dot-org/blob/600dfa123922d9ee43dfceb71d7197891be263f4/dashboard/app/models/ability.rb#L274-L276

I believe you'll also want to add this line at the top of the class as well, which will make it so that we'll check ability.rb to see whether a user has permission for a given controller action:

https://github.com/code-dot-org/code-dot-org/blob/600dfa123922d9ee43dfceb71d7197891be263f4/dashboard/app/controllers/javabuilder_sessions_controller.rb#L7

Reply:
In general, the `javabuilder_sessions_controller` might be a good example to follow here (with testing examples as well), as it does something similar (makes an internal POST request to javabuilder)